api_app/services/azure_resource_status.py

```diff
@@ -14,7 +14,11 @@ def get_azure_resource_status(resource_id):
     try:
         if resource_type == 'Microsoft.Compute/virtualMachines':
             vm_instance_view: models.VirtualMachineInstanceView = get_azure_vm_instance_view(resource_name, resource_group_name)
-            power_state = [x for x in vm_instance_view.statuses if x.code.startswith('PowerState')][0].display_status
+            power_state = None
+            if vm_instance_view.statuses is not None:
```

Comment:
I think you could have done something like...

```
[k for k in (tags or [])]
```

... but being more verbose is probably better here so ignore this.

Reply:
Oooh, that's good to know. That line is already has a reasonable amount of complexity, so I agree that verbose works well here.